addons/mail/static/src/messaging.js

```diff
@@ -142,6 +142,7 @@ export class Messaging {
             chatPartnerId: false,
             isAdmin: false,
             canLeave: data.canLeave || false,
+            isRenameable: ["chat", "channel", "group"].includes(type),
```

Comment:
Ok, should keep in mind it's computed only at creation of thread.

I saw another PR that was preparing an `updateThread({ data })`, which might have received another `type` and kept the old `isRenameable` value because was not recomputed.

Thankfully here, we don't have functional use of changing the type of a thread. But it's quite important to see the subtle assumptions we make in the code. A small change of functional specs or cleaning of modelling can very easy break the code because of this state that we should imperatively maintain.

It's easy to see it right now because it's in the diff of this PR. Later on can be hard to spot if similar kind of assumptions is spread everywhere in the code.

Reply:
Yes, I think for now it is still ok to have this field computed only at the creation due to we don't allow changing the type of the thread. Otherwise, it will be annoying to have a recompute on that every time we want to invoke the `updateThread` and we better have a check on the `data` provided to the `updateThread` to get avoid changing the type which we don't want to.
When we want to allow the type change of the thread at some time, we can do the accordingly change at that time I suppose.